### ISSUE TITLE
broken integer narrowing

### ISSUE THREAD

**Issue by sekiguchi-nagisa** [sekiguchi-nagisa/ydsh#175]
## Impact Target
*  v0.3.0
*  v0.4.0-unstable

## Steps
```
ydsh> var a = -1 as Byte
ydsh> $a
(Byte) 255
ydsh> assert($a == 255)
Assertion Error:
    from (stdin):3 '<toplevel>()'
```

## Problem
assertion failure

## Expect
assertion success

## Note
need to fix cast node

**Comment by sekiguchi-nagisa** [comment]
* if same precision (ex. Int16 to Uint16)

   do nothing
*  if to higher precision
    
   do nothing

**Comment by sekiguchi-nagisa** [comment]
test with comparing stringified value

**Comment by sekiguchi-nagisa** [comment]
fix number cast code in commit c0b0e8efc866096b032b5fb7d28e5278496bc001

**Comment by sekiguchi-nagisa** [comment]
add test case in commit 64c6cbf38e9bc3e39dcf838c874417c572603d4f to 8d411a5c7c08521a247469aa1427a64033622294
